src/Integration/WPCOMProxy.php

```diff
@@ -0,0 +1,215 @@
+	}
+
+	/**
+	 * Check if a single item is syncable.
+	 *
+	 * @param WP_REST_Response $response The response object.
+	 * @param mixed            $item     The item.
+	 * @param WP_REST_Request  $request  The request object.
+	 *
+	 * @return WP_REST_Response The response object updated.
+	 */
+	public function check_item_is_syncable( $response, $item, WP_REST_Request $request ): WP_REST_Response {
+		if ( ! $this->should_filter_data( $request ) ) {
+			return $response;
+		}
+
+		$route               = $request->get_route();
+		$pattern             = '/(?P<resource>[\w]+)\/(?P<id>[\d]+$)/';
+		$protected_resources = [
+			'products',
+			'coupons',
+		];
+		preg_match( $pattern, $route, $matches );
+
+		if ( ! isset( $matches['id'] ) || ! isset( $matches['resource'] ) || ! in_array( $matches['resource'], $protected_resources, true ) ) {
+			return $response;
+		}
+
+		$meta_data = $response->get_data()['meta_data'] ?? [];
+
+		foreach ( $meta_data as $meta ) {
+			if ( $meta->key === '_wc_gla_visibility' && $meta->value === 'sync-and-show' ) {
+				return $response;
+			}
+		}
+
+		return new WP_REST_Response(
+			[
+				'code'    => 'gla_rest_item_no_syncable',
+				'message' => 'Item not syncable',
+				'data'    => [
+					'status' => '403',
+				],
+			],
+			403
+		);
+	}
+
+	/**
+	 * Query items with specific args for example where _wc_gla_visibility is set to sync-and-show.
+	 *
+	 * @param array           $args    The query args.
+	 * @param WP_REST_Request $request The request object.
+	 *
+	 * @return array The query args updated.
+	 * */
+	public function filter_by_metaquery( array $args, WP_REST_Request $request ): array {
+		if ( ! $this->should_filter_data( $request ) ) {
+			return $args;
+		}
+
+		$post_type         = $args['post_type'];
+		$post_type_filters = self::$post_types_to_filter[ $post_type ];
+
+		if ( ! isset( $post_type_filters['meta_query'] ) || ! is_array( $post_type_filters['meta_query'] ) ) {
+			return $args;
+		}
+
+		$args['meta_query'] = [ ...$args['meta_query'] ?? [], ...$post_type_filters['meta_query'] ];
+
+		return $args;
+	}
+
+	/**
+	 * Filter the metadata of an object.
+	 *
+	 * @param WP_REST_Response $response The response object.
+	 * @param mixed            $item     The item.
+	 * @param WP_REST_Request  $request  The request object.
+	 *
+	 * @return WP_REST_Response The response object updated.
+	 */
+	public function filter_metadata( WP_REST_Response $response, $item, WP_REST_Request $request ): WP_REST_Response {
+		if ( ! $this->should_filter_data( $request ) ) {
+			return $response;
+		}
+
+		$data = $response->get_data();
+
+		if ( ! isset( $data['meta_data'] ) ) {
+			return $response;
+		}
+
+		foreach ( $data['meta_data'] as $key => $meta ) {
+			if ( str_starts_with( $meta->key, '_' ) && ! str_starts_with( $meta->key, '_wc_gla' ) ) {
+				unset( $data['meta_data'][ $key ] );
+			}
+		}
+
+		$data['meta_data'] = array_values( $data['meta_data'] );
```

Comment:
Why we are just returning the values?

Reply:
Thanks for catching this, I believe it was there because I used `array_filter` to return values and I wanted to reset the indexes. However, it's no longer necessary. I removed here: https://github.com/woocommerce/google-listings-and-ads/pull/2152/commits/5ec640eed1f815a9e83cbfe0a4832b6e9627bce1